tests/handlers/test_koji.py

```diff
@@ -48,6 +48,143 @@ def test_build_complete(mock_koji_cs, mock_getBuild_two):
+
+    commit = DistGitCommit.nodes.get_or_none(hash_='09d40c9bdfd34c5130f8f02e49e059efd33bddf7')
+
+    build.commit.is_connected(commit)
+
+
+@mock.patch('koji.ClientSession')
+def test_build_failed(mock_koji_cs, mock_getBuild_four, kb_four):
+    """Test the Koji handler when it recieves a new build failed message."""
+    mock_koji_session = mock.Mock()
+    mock_koji_session.getBuild.return_value = mock_getBuild_four
+    mock_koji_cs.return_value = mock_koji_session
+
+    orig_build = kb_four
+    assert orig_build.state == 0
+
+    with open(path.join(message_dir, 'koji', 'build_failed.json'), 'r') as f:
+        msg = json.load(f)
+    assert KojiHandler.can_handle(msg) is True
+    handler = KojiHandler(config)
+    handler.handle(msg)
+
+    build = KojiBuild.nodes.get_or_none(id_='754575')
+    assert build is not None
+    assert build.completion_time == datetime(2018, 6, 15, 20, 26, 38, tzinfo=pytz.utc)
+    assert build.creation_time == datetime(2018, 8, 20, 17, 32, 24, tzinfo=pytz.utc)
+    assert build.epoch is None
+    assert build.extra == ('{"source": {"original_url": "git://pkgs.domain.com/rpms/podman#c75628'
+                           'd32fc0c8b1bd5e1e7b047538a25c983569"}}')
+    assert build.id_ == '754575'
+    assert build.name == 'podman'
+    assert build.release == '1.git9d09a4d.el8'
+    assert build.version == '0.8.3'
+    assert build.start_time == datetime(2018, 8, 20, 17, 32, 24, tzinfo=pytz.utc)
+    assert build.state == 3
+
+    commit = DistGitCommit.nodes.get_or_none(hash_='c75628d32fc0c8b1bd5e1e7b047538a25c983569')
+
+    build.commit.is_connected(commit)
+
+
+@mock.patch('koji.ClientSession')
+def test_build_canceled(mock_koji_cs, mock_getBuild_canceled, kb_canceled):
+    """Test the Koji handler when it recieves a new build failed message."""
+    mock_koji_session = mock.Mock()
+    mock_koji_session.getBuild.return_value = mock_getBuild_canceled
+    mock_koji_cs.return_value = mock_koji_session
+
+    orig_build = kb_canceled
+    assert orig_build.state == 4
+
+    with open(path.join(message_dir, 'koji', 'build_canceled.json'), 'r') as f:
+        msg = json.load(f)
+    assert KojiHandler.can_handle(msg) is True
+    handler = KojiHandler(config)
+    handler.handle(msg)
+
+    build = KojiBuild.nodes.get_or_none(id_='751869')
+    assert build is not None
+    assert build.completion_time == datetime(2018, 8, 18, 15, 4, 47, tzinfo=pytz.utc)
+    assert build.creation_time == datetime(2018, 8, 15, 13, 47, 43, tzinfo=pytz.utc)
+    assert build.epoch is None
+    assert build.extra == ('{"source": {"original_url": "git://pkgs.domain.com/rpms/gcc#c72278'
+                           '1ead329f9b76751ba89ee7451109cebde6"}}')
+    assert build.id_ == '751869'
+    assert build.name == 'devtoolset-8-gcc'
+    assert build.release == '2.2.el7'
+    assert build.version == '8.2.1'
+    assert build.start_time == datetime(2018, 8, 15, 13, 47, 43, tzinfo=pytz.utc)
+    assert build.state == 4
```

Comment:
The build in Neo4j was already cancelled, so this test isn't actually testing the state transitioned. Could you use pytest fixture that creates a build in the build state in Neo4j instead?

Reply:
Optional: Since you are really just checking the state transition occurred, you can remove all these other asserts above.